File: src/ObjectManager/Object.Tes/FilePacks/TesDataPack.cs
using OA.Core;
using System.IO;
using UnityEngine;

namespace OA.Tes.FilePacks
{
    public class TesDataPack : EsmFile, IDataPack
    {
        string _webPath;

        public TesDataPack(string filePath, string webPath, GameId gameId)
            : base(filePath != null && File.Exists(filePath) ? filePath : null, gameId)
        {
            _webPath = webPath;
        }

        ICellRecord IDataPack.FindCellRecord(Vector3Int cellId) { return FindCellRecord(cellId); }
        //ICellRecord IDataPack.FindInteriorCellRecord(string cellId) { return FindInteriorCellRecord(cellName); }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/TesDataPack.swift
//
//  TesDataPack.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

import Foundation
import simd

public class TesDataPack: EsmFile, IDataPack {
    override init(_ filePath: URL?, for game: GameId) {
        let fileManager = FileManager.default
        super.init(filePath != nil && fileManager.fileExists(atPath: filePath!.path) ? filePath : nil, for: game)
    }
    
    public func findCellRecord(_ cellId: int3) -> ICellRecord? {
        return super.findCellRecord(cellId: cellId)
    }
}
